compiler/rustc_errors/src/emitter.rs

```diff
@@ -644,6 +644,8 @@ impl EmitterWriter {
         code_offset: usize,
         margin: Margin,
     ) {
+        // Tabs are assumed to have been replaced by spaces in calling code.
+        assert!(!source_string.contains('\t'));
```

Comment:
This is a bit heavy handed but ok with me (I will assume we'll see reports on nightly if this causes issues before it hits stable).

Reply:
Ah okay, mostly I just wanted it to be ultra clear to future code readers which paths should not have tabs, but it could be relaxed to a comment of course. I'll check `json` output now.